src/Qir/Execution/Tests.Microsoft.Quantum.Qir.Tools/QirExecutableTests.cs

```diff
@@ -0,0 +1,119 @@
+        private readonly IList<DirectoryInfo> headerDirectories;
+        private readonly IList<DirectoryInfo> libraryDirectories;
+
+        public QirExecutableTests()
+        {
+            // Set up files.
+            var prefix = Guid.NewGuid().ToString();
+            binDirectory = new DirectoryInfo($"{prefix}-bin");
+            binDirectory.Create();
+            libraryDirectory = new DirectoryInfo($"{prefix}-library");
+            libraryDirectory.Create();
+            includeDirectory = new DirectoryInfo($"{prefix}-include");
+            includeDirectory.Create();
+            sourceDirectory = new DirectoryInfo($"{prefix}-source");
+            sourceDirectory.Create();
+            executableFile = new FileInfo(Path.Combine(binDirectory.FullName, "executable"));
+            driverGeneratorMock = new Mock<IQirDriverGenerator>();
+            executableGeneratorMock = new Mock<IQirExecutableGenerator>();
+            runnerMock = new Mock<IQuantumExecutableRunner>();
+            qirExecutable = new Mock<QirExecutable>(executableFile, qirBytecode, Mock.Of<ILogger>(), driverGeneratorMock.Object, executableGeneratorMock.Object, runnerMock.Object) { CallBase = true };
+            linkLibraries = new List<string> { "lib1", "lib2" };
+            headerDirectories = new List<DirectoryInfo>();
+            libraryDirectories = new List<DirectoryInfo>();
+            qirExecutable.SetupGet(obj => obj.LinkLibraries).Returns(linkLibraries);
+            qirExecutable.SetupGet(obj => obj.SourceDirectoryPath).Returns(sourceDirectory.FullName);
+            qirExecutable.SetupGet(obj => obj.DriverFileExtension).Returns(".cpp");
+            qirExecutable.SetupGet(obj => obj.HeaderDirectories).Returns(headerDirectories);
+            qirExecutable.SetupGet(obj => obj.LibraryDirectories).Returns(libraryDirectories);
+        }
+
+        public void Dispose()
+        {
+            Util.DeleteDirectory(sourceDirectory);
+            Util.DeleteDirectory(includeDirectory);
+            Util.DeleteDirectory(libraryDirectory);
+            Util.DeleteDirectory(binDirectory);
+        }
+
+        [Fact]
+        public async Task TestBuild()
+        {
+            // Set up.
+            var entryPoint = new EntryPointOperation();
+            var driverFileContents = "driver file contents";
+            driverGeneratorMock.Setup(obj => obj.GenerateAsync(entryPoint, It.IsAny<Stream>())).Callback<EntryPointOperation, Stream>((entryPoint, stream) =>
+            {
+                using var streamWriter = new StreamWriter(stream);
+                streamWriter.Write(driverFileContents);
+            });
+
+            // Build the executable.
+            await qirExecutable.Object.BuildAsync(entryPoint, new[] { libraryDirectory }, new[] { includeDirectory });
+
+            // Verify that the "bytecode" file was created correctly.
+            var bytecodeFilePath = new FileInfo(Path.Combine(sourceDirectory.FullName, "qir.bc"));
```

Comment:
The following may be a bit faster for a typical case (when files are equal) as it doesn't need to invoke assert for every byte.
            byte[] bytecodeFileContent = File.ReadAllBytes(Path.Combine(sourceDirectory.FullName, "qir.bc"));
            Assert.True(qirBytecode.SequenceEqual(bytecodeFileContent));

Reply:
Created a comparison method in Utils.cs and changed this to `Assert.True(Util.CompareStreams(new MemoryStream(qirBytecode), bytecodeFileStream));`